ui/selectors/multichain-accounts/feature-flags.ts

```diff
@@ -0,0 +1,109 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+// We can ignore the unused vars warning while the flag is not active
+
+import {
+  getRemoteFeatureFlags,
+  type RemoteFeatureFlagsState,
+} from '../remote-feature-flags';
+
+/**
+ * Feature flag type for multichain accounts features
+ */
+export type MultichainAccountsFeatureFlag = {
+  enabled: boolean;
+  featureVersion: string | null;
+  minimumVersion: string | null;
+};
+
+// TODO: Update the value to the decided version multichain accounts will be released
+const MINIMUM_SUPPORTED_VERSION = null;
+const FEATURE_VERSION_1 = '1';
+const FEATURE_VERSION_2 = '2';
+
+/**
+ * Compare is the version1 is greater than or equal to version2
+ *
+ * @param version1 - The first version string to compare.
+ * @param version2 - The second version string to compare.
+ * @returns boolean - True if version1 is greater than or equal to version2, false otherwise.
+ */
+export const compareVersions = (version1: string, version2: string) => {
+  const regex = /^\d+\.\d+\.\d+$/u;
+  if (!regex.test(version1) || !regex.test(version2)) {
+    return false; // Invalid version format
+  }
+
+  const v1 = version1.split('.').map(Number);
+  const v2 = version2.split('.').map(Number);
+
+  for (let i = 0; i < Math.max(v1.length, v2.length); i++) {
+    const num1 = v1[i] || 0;
+    const num2 = v2[i] || 0;
+
+    if (num1 > num2) {
+      return true;
+    }
+    if (num1 < num2) {
+      return false;
+    }
+  }
+
+  return true; // Versions are equal
+};
```

Comment:
WDYT about relying on `semver` package instead here? Using `semver.valid` and `semver.gte` would do the trick I guess.

Also, I think that we should `console.warn` if any of the versions are invalid. This should not crash the whole app, but logging it somewhere could help catch some bad configs if any.

Reply:
good to know, I was not aware we had this dependency

Reply:
done in https://github.com/MetaMask/metamask-extension/pull/33112/commits/a97efefadf7f895e14fa67d6dee87bebd78f1235